p4rt_app/p4rt.cc

```diff
@@ -17,101 +17,394 @@
+
+    p4rt_app::ConfigDbNodeCfgTableEventHandler node_table_handler(
+        &p4runtime_server);
+    auto node_table_status =
+        config_db_monitor.RegisterTableHandler("NODE_CFG", node_table_handler);
+    if (!node_table_status.ok()) {
+      LOG(FATAL)  // Crash OK: only fails on startup.
+          << "CONFIG_DB event monitor could not register 'NODE_CFG': "
+          << node_table_status;
+    }
+
+    swss::DBConnector port_events_app_db("APPL_DB", /*timeout=*/0);
+    swss::DBConnector port_events_app_state_db("APPL_STATE_DB", /*timeout=*/0);
+    p4rt_app::ConfigDbPortTableEventHandler port_table_handler(
+        &p4runtime_server,
+        std::make_unique<p4rt_app::sonic::TableAdapter>(&port_events_app_db,
+                                                        "P4RT_PORT_ID_TABLE"),
+        std::make_unique<p4rt_app::sonic::TableAdapter>(
+            &port_events_app_state_db, "P4RT_PORT_ID_TABLE"));
+    auto port_table_status =
+        config_db_monitor.RegisterTableHandler("PORT", port_table_handler);
+    if (!port_table_status.ok()) {
+      LOG(FATAL)  // Crash OK: only fails on startup.
+          << "CONFIG_DB event monitor could not register 'PORT': "
+          << port_table_status;
+    }
+
+    auto port_channel_table_status = config_db_monitor.RegisterTableHandler(
+        "PORTCHANNEL", port_table_handler);
+    if (!port_channel_table_status.ok()) {
+      LOG(FATAL)  // Crash OK: only fails on startup.
+          << "CONFIG_DB event monitor could not register 'PORTCHANNEL': "
+          << port_channel_table_status;
+    }
+
+    while (monitor_config_db_events) {
+      absl::Status status = config_db_monitor.WaitForNextEventAndHandle();
+      if (!status.ok()) {
+        LOG(ERROR) << "CONFIG_DB event monitor failed waiting for an event: "
+                   << status;
+      }
+    }
+  });
+
+  // Start listening for state verification events, and update StateDb for P4RT.
+  swss::DBConnector state_verification_db(STATE_DB, kRedisDbHost, kRedisDbPort,
+                                          /*timeout=*/0);
+  p4rt_app::sonic::TableAdapter state_verification_table_adapter(
+      &state_verification_db, "VERIFY_STATE_RESP_TABLE");
+  p4rt_app::sonic::ConsumerNotifierAdapter state_verification_notifier(
+      "VERIFY_STATE_REQ_CHANNEL", &state_verification_db);
+  p4rt_app::StateVerificationEvents state_verification_event_monitor(
+      p4runtime_server, state_verification_notifier,
+      state_verification_table_adapter);
+  state_verification_event_monitor.Start();
 
   // Start a P4 runtime server
   ServerBuilder builder;
+  auto server_cred = BuildServerCredentials();
+  if (!server_cred.ok()) {
+    LOG(ERROR) << "Failed to build server credentials, error "
+               << server_cred.status();
+    return -1;
+  }
+
   std::string server_addr = absl::StrCat("[::]:", FLAGS_p4rt_grpc_port);
-  builder.AddListeningPort(server_addr, grpc::InsecureServerCredentials());
```

Comment:
Do we need additional doc on the certificates to make other upstream users be able to get the right files ?

Reply:
We did add extra documentation about setting these flags: http://go/gh/sonic-net/SONiC/pull/1054. However, by default the certs are shared between p4rt and the telmetry code so most users should be okay.